examples/basics/plotting/plot.py

```diff
@@ -35,25 +35,21 @@
         l.update_gl_state(depth_test=False)
 
 labelgrid = fig[0, 0].view.add_grid(margin=10)
-hspacer = vp.Widget()
-hspacer.stretch = (6, 1)
-labelgrid.add_widget(hspacer, row=0, col=0)
 
 box = vp.Widget(bgcolor=(1, 1, 1, 0.6), border_color='k')
-labelgrid.add_widget(box, row=0, col=1)
+box_widget = labelgrid.add_widget(box, row=0, col=1)
+box_widget.width = 90
+box_widget.height = 120
 
-vspacer = vp.Widget()
-vspacer.stretch = (1, 2)
-labelgrid.add_widget(vspacer, row=1, col=1)
+bottom_spacer = vp.Widget()
+labelgrid.add_widget(bottom_spacer, row=1, col=0)
 
 labels = [vp.Label('n=%d' % plot_nvals[i], color=colors[i], anchor_x='left')
           for i in range(len(plot_nvals))]
 boxgrid = box.add_grid()
 for i, label in enumerate(labels):
-    boxgrid.add_widget(label, row=i, col=0)
-hspacer2 = vp.Widget()
-hspacer2.stretch = (4, 1)
-boxgrid.add_widget(hspacer2, row=0, col=1)
+    label_widget = boxgrid.add_widget(label, row=i, col=0)
+
```

Comment:
Ah yes, making them private was on my todo list :)

There's a contract that I'm not sure how to express properly - the
GridWidget uses the var_* of its children, even though the var_* are
private.

In C++, I could use "friend" to express that relationship. I'm not sure
what the right was is in Python

On Thu, 7 Jan 2016 at 12:21 Luke Campagnola <notifications@github.com>
wrote:

> In examples/basics/plotting/plot.py
> <https://github.com/vispy/vispy/pull/1049#discussion_r49046052>:
>
> >
> >  labels = [vp.Label('n=%d' % plot_nvals[i], color=colors[i], anchor_x='left')
> >            for i in range(len(plot_nvals))]
> >  boxgrid = box.add_grid()
> >  for i, label in enumerate(labels):
> > -    boxgrid.add_widget(label, row=i, col=0)
> > -hspacer2 = vp.Widget()
> > -hspacer2.stretch = (4, 1)
> > -boxgrid.add_widget(hspacer2, row=0, col=1)
> > +    label_widget = boxgrid.add_widget(label, row=i, col=0)
> > +
>
> Bump
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vispy/vispy/pull/1049/files#r49046052>.
>